<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="simplifyEnrichment">
<title>Summarize simplifyEnrichment plots â€¢ simplifyEnrichment</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Summarize simplifyEnrichment plots">
<meta property="og:description" content="simplifyEnrichment">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">simplifyEnrichment</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/simplifyEnrichment_intro.html">1. Simplify Functional Enrichment Results</a>
    <a class="dropdown-item" href="../articles/summarize.html">2. Summarize simplifyEnrichment plots</a>
    <a class="dropdown-item" href="../articles/interactive.html">3. A Shiny app to interactively visualize clustering results</a>
    <a class="dropdown-item" href="../articles/word_cloud_anno.html">4. Word Cloud Annotation</a>
  </div>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://jokergoo.github.io/simplifyEnrichment_supplementary/">Supplementary</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jokergoo/simplifyEnrichment/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Summarize simplifyEnrichment plots</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu (<a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a>)</h4>
            
            <h4 data-toc-skip class="date">2024-09-13</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jokergoo/simplifyEnrichment/blob/HEAD/vignettes/summarize.Rmd" class="external-link"><code>vignettes/summarize.Rmd</code></a></small>
      <div class="d-none name"><code>summarize.Rmd</code></div>
    </div>

    
    
<p>The <code>lt</code> object contains a list of GO enrichment tables.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">lt</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "BP_km1" "BP_km2" "BP_km3" "BP_km4"</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">lt</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    ID                              Description GeneRatio   BgRatio       pvalue</span></span>
<span><span class="co">## GO:0006974 GO:0006974 cellular response to DNA damage stimulus    77/471 871/18614 3.252635e-22</span></span>
<span><span class="co">## GO:0006281 GO:0006281                               DNA repair    61/471 587/18614 3.683205e-21</span></span>
<span><span class="co">## GO:0000278 GO:0000278                       mitotic cell cycle    78/471 933/18614 5.191864e-21</span></span>
<span><span class="co">## GO:1903047 GO:1903047               mitotic cell cycle process    70/471 777/18614 1.148628e-20</span></span>
<span><span class="co">## GO:0051276 GO:0051276                  chromosome organization    58/471 626/18614 8.913192e-18</span></span>
<span><span class="co">## GO:0006260 GO:0006260                          DNA replication    38/471 278/18614 1.862901e-17</span></span>
<span><span class="co">##                p.adjust       qvalue</span></span>
<span><span class="co">## GO:0006974 1.545002e-18 1.174030e-18</span></span>
<span><span class="co">## GO:0006281 8.220452e-18 6.246633e-18</span></span>
<span><span class="co">## GO:0000278 8.220452e-18 6.246633e-18</span></span>
<span><span class="co">## GO:1903047 1.363995e-17 1.036485e-17</span></span>
<span><span class="co">## GO:0051276 8.467532e-15 6.434386e-15</span></span>
<span><span class="co">## GO:0006260 1.474797e-14 1.120682e-14</span></span></code></pre>
<p>We first demonstrate the new plot on the single enrichment table. To use the <strong>simplifyEnrichment</strong> package, we extract significant GO terms, and then call <code><a href="../reference/simplifyGO.html">simplifyGO()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/simplifyEnrichment" class="external-link">simplifyEnrichment</a></span><span class="op">)</span></span>
<span><span class="va">df</span> <span class="op">=</span> <span class="va">lt</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">go_id</span> <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span><span class="va">df</span><span class="op">$</span><span class="va">p.adjust</span> <span class="op">&lt;</span> <span class="fl">0.01</span><span class="op">]</span></span>
<span><span class="fu"><a href="../reference/simplifyGO.html">simplifyGO</a></span><span class="op">(</span><span class="va">go_id</span><span class="op">)</span></span></code></pre></div>
<p><img src="summarize_files/figure-html/unnamed-chunk-4-1.png" width="806.4" style="display: block; margin: auto;"></p>
<p>The plot looks good, but it still contains too many graphic contents. For example, the GO similarity heatmap is useful, but it takes too much space on the final plot. Here I developed a new function <code><a href="../reference/summarizeGO.html">summarizeGO()</a></code> which simplifies the enrichment results even more. The idea is that since we already have the GO clusters, with a certain statistic of enrichment, we can simply use its average for the GO cluster.</p>
<p>In the following example, we use <code>-log10(p.adjust)</code> as an enrichment measure. The heights of bars correspond to the mean of <code>-log10(p.adjust)</code> of GO terms in different GO clusters. On the left side, we still use the word clouds which efficiently show the general functions in each GO cluster.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">l</span> <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">p.adjust</span> <span class="op">&lt;</span> <span class="fl">0.01</span></span>
<span><span class="fu"><a href="../reference/summarizeGO.html">summarizeGO</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span><span class="va">l</span><span class="op">]</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">p.adjust</span><span class="op">)</span><span class="op">[</span><span class="va">l</span><span class="op">]</span>, axis_label <span class="op">=</span> <span class="st">"average -log10(p.adjust)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="summarize_files/figure-html/unnamed-chunk-5-1.png" width="691.2" style="display: block; margin: auto;"></p>
<p>GO IDs can be attached to the numeric value vector, but this time, the <code>value</code> argument should be explicitely specified when calling <code><a href="../reference/summarizeGO.html">summarizeGO()</a></code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">v</span> <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">p.adjust</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span> <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">ID</span></span>
<span><span class="fu"><a href="../reference/summarizeGO.html">summarizeGO</a></span><span class="op">(</span>value <span class="op">=</span> <span class="va">v</span><span class="op">[</span><span class="va">l</span><span class="op">]</span>, axis_label <span class="op">=</span> <span class="st">"average -log10(p.adjust)"</span><span class="op">)</span></span></code></pre></div>
<p>Beside <code>-log10(p.adjust)</code>, we also suggest to use log2 fold enrichment as the enrichment measure. It is calculated as</p>
<p><span class="math display">\[ \log_2 \left( \frac{k/m_1}{m_2/n} \right) \]</span></p>
<p>where <span class="math inline">\(k\)</span> is the number of DE genes (if the genes of interest are DE genes) in a gene set, <span class="math inline">\(m_1\)</span> is the size of DE genes, <span class="math inline">\(m_2\)</span> is the size of the gene set, <span class="math inline">\(n\)</span> is the total number of genes in the universal set. Of course, the definition of <span class="math inline">\(m_1\)</span> and <span class="math inline">\(m_2\)</span> can be switched.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">k</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"/\\d+$"</span>, <span class="st">""</span>, <span class="va">df</span><span class="op">$</span><span class="va">GeneRatio</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">m1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"^\\d+/"</span>, <span class="st">""</span>, <span class="va">df</span><span class="op">$</span><span class="va">GeneRatio</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">m2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"/\\d+$"</span>, <span class="st">""</span>, <span class="va">df</span><span class="op">$</span><span class="va">BgRatio</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"^\\d+/"</span>, <span class="st">""</span>, <span class="va">df</span><span class="op">$</span><span class="va">BgRatio</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">log2_fold_enrichment</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">k</span><span class="op">*</span><span class="va">n</span><span class="op">/</span><span class="va">m1</span><span class="op">/</span><span class="va">m2</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/summarizeGO.html">summarizeGO</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span><span class="va">l</span><span class="op">]</span>, <span class="va">log2_fold_enrichment</span><span class="op">[</span><span class="va">l</span><span class="op">]</span>, axis_label <span class="op">=</span> <span class="st">"average log2(fold_enrichment)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="summarize_files/figure-html/unnamed-chunk-7-1.png" width="691.2" style="display: block; margin: auto;"></p>
<p>Of course, you can construct a named <code>log2_fold_enrichment</code> vector which only contains significant GO terms.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">log2_fold_enrichment</span><span class="op">)</span> <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">ID</span></span>
<span><span class="fu"><a href="../reference/summarizeGO.html">summarizeGO</a></span><span class="op">(</span>value <span class="op">=</span> <span class="va">log2_fold_enrichment</span><span class="op">[</span><span class="va">l</span><span class="op">]</span>, axis_label <span class="op">=</span> <span class="st">"average log2(fold_enrichment)"</span><span class="op">)</span></span></code></pre></div>
<p>For multiple GO enrichment results, <code><a href="../reference/simplifyGOFromMultipleLists.html">simplifyGOFromMultipleLists()</a></code> can be used to visualize and compare GO clusters.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/simplifyGOFromMultipleLists.html">simplifyGOFromMultipleLists</a></span><span class="op">(</span><span class="va">lt</span>, padj_cutoff <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></span></code></pre></div>
<p><img src="summarize_files/figure-html/unnamed-chunk-9-1.png" width="100%" style="display: block; margin: auto;"></p>
<p><code><a href="../reference/summarizeGO.html">summarizeGO()</a></code> can also be used to simplify such plot. Now the value of <code>value</code> is a list of numeric named vectors which contains significant GO terms in each enrichment table:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">value</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">lt</span>, <span class="kw">function</span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">v</span> <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">p.adjust</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span> <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">ID</span></span>
<span>    <span class="va">v</span><span class="op">[</span><span class="va">df</span><span class="op">$</span><span class="va">p.adjust</span> <span class="op">&lt;</span> <span class="fl">0.001</span><span class="op">]</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/summarizeGO.html">summarizeGO</a></span><span class="op">(</span>value <span class="op">=</span> <span class="va">value</span>, axis_label <span class="op">=</span> <span class="st">"average -log10(p.adjust)"</span>, legend_title <span class="op">=</span> <span class="st">"-log10(p.adjust)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="summarize_files/figure-html/unnamed-chunk-10-1.png" width="691.2" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.3 (2024-02-29)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Sonoma 14.6.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Random number generation:</span></span>
<span><span class="co">##  RNG:     L'Ecuyer-CMRG </span></span>
<span><span class="co">##  Normal:  Inversion </span></span>
<span><span class="co">##  Sample:  Rejection </span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Europe/Berlin</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] simplifyEnrichment_1.99.0 hu6800.db_3.13.0          org.Hs.eg.db_3.17.0      </span></span>
<span><span class="co">##  [4] AnnotationDbi_1.62.2      IRanges_2.36.0            S4Vectors_0.40.2         </span></span>
<span><span class="co">##  [7] Biobase_2.60.0            BiocGenerics_0.48.1       cola_2.6.0               </span></span>
<span><span class="co">## [10] knitr_1.45               </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] splines_4.3.3           later_1.3.2             bitops_1.0-7           </span></span>
<span><span class="co">##   [4] ggplotify_0.1.2         tibble_3.2.1            polyclip_1.10-6        </span></span>
<span><span class="co">##   [7] XML_3.99-0.16.1         lifecycle_1.0.4         doParallel_1.0.17      </span></span>
<span><span class="co">##  [10] NLP_0.2-1               lattice_0.22-5          MASS_7.3-60.0.1        </span></span>
<span><span class="co">##  [13] magrittr_2.0.3          sass_0.4.9              rmarkdown_2.26         </span></span>
<span><span class="co">##  [16] jquerylib_0.1.4         yaml_2.3.8              httpuv_1.6.15          </span></span>
<span><span class="co">##  [19] doRNG_1.8.6             cowplot_1.1.3           DBI_1.2.2              </span></span>
<span><span class="co">##  [22] RColorBrewer_1.1-3      eulerr_7.0.2            zlibbioc_1.46.0        </span></span>
<span><span class="co">##  [25] purrr_1.0.2             ggraph_2.2.1            RCurl_1.98-1.14        </span></span>
<span><span class="co">##  [28] yulab.utils_0.1.4       tweenr_2.0.3            circlize_0.4.16        </span></span>
<span><span class="co">##  [31] GenomeInfoDbData_1.2.10 enrichplot_1.20.3       tm_0.7-13              </span></span>
<span><span class="co">##  [34] ggrepel_0.9.5           irlba_2.3.5.1           tidytree_0.4.6         </span></span>
<span><span class="co">##  [37] genefilter_1.82.1       annotate_1.78.0         brew_1.0-10            </span></span>
<span><span class="co">##  [40] commonmark_1.9.1        pkgdown_2.0.9           codetools_0.2-19       </span></span>
<span><span class="co">##  [43] DOSE_3.26.2             xml2_1.3.6              ggforce_0.4.2          </span></span>
<span><span class="co">##  [46] tidyselect_1.2.1        shape_1.4.6.1           aplot_0.2.2            </span></span>
<span><span class="co">##  [49] farver_2.1.1            viridis_0.6.5           matrixStats_1.3.0      </span></span>
<span><span class="co">##  [52] jsonlite_1.8.8          GetoptLong_1.0.5        tidygraph_1.3.1        </span></span>
<span><span class="co">##  [55] survival_3.5-8          iterators_1.0.14        systemfonts_1.0.6      </span></span>
<span><span class="co">##  [58] foreach_1.5.2           tools_4.3.3             treeio_1.24.3          </span></span>
<span><span class="co">##  [61] ragg_1.3.1              Rcpp_1.0.12             glue_1.7.0             </span></span>
<span><span class="co">##  [64] gridExtra_2.3           xfun_0.43               qvalue_2.32.0          </span></span>
<span><span class="co">##  [67] flexclust_1.4-2         MatrixGenerics_1.12.3   GenomeInfoDb_1.36.4    </span></span>
<span><span class="co">##  [70] dplyr_1.1.4             withr_3.0.0             fastmap_1.1.1          </span></span>
<span><span class="co">##  [73] fansi_1.0.6             digest_0.6.35           R6_2.5.1               </span></span>
<span><span class="co">##  [76] mime_0.12               gridGraphics_0.5-1      microbenchmark_1.4.10  </span></span>
<span><span class="co">##  [79] textshaping_0.3.7       colorspace_2.1-0        Cairo_1.6-2            </span></span>
<span><span class="co">##  [82] GO.db_3.17.0            markdown_1.12           RSQLite_2.3.6          </span></span>
<span><span class="co">##  [85] utf8_1.2.4              tidyr_1.3.1             generics_0.1.3         </span></span>
<span><span class="co">##  [88] data.table_1.15.4       class_7.3-22            graphlayouts_1.1.1     </span></span>
<span><span class="co">##  [91] httr_1.4.7              htmlwidgets_1.6.4       scatterpie_0.2.2       </span></span>
<span><span class="co">##  [94] scatterplot3d_0.3-44    pkgconfig_2.0.3         gtable_0.3.5           </span></span>
<span><span class="co">##  [97] modeltools_0.2-23       blob_1.2.4              ComplexHeatmap_2.18.0  </span></span>
<span><span class="co">## [100] impute_1.74.1           XVector_0.40.0          clusterProfiler_4.8.3  </span></span>
<span><span class="co">## [103] shadowtext_0.1.3        htmltools_0.5.8.1       fgsea_1.26.0           </span></span>
<span><span class="co">## [106] clue_0.3-65             scales_1.3.0            png_0.1-8              </span></span>
<span><span class="co">## [109] ggfun_0.1.4             reshape2_1.4.4          rjson_0.2.21           </span></span>
<span><span class="co">## [112] nlme_3.1-164            cachem_1.0.8            GlobalOptions_0.1.2    </span></span>
<span><span class="co">## [115] Polychrome_1.5.1        stringr_1.5.1           parallel_4.3.3         </span></span>
<span><span class="co">## [118] HDO.db_0.99.1           desc_1.4.3              pillar_1.9.0           </span></span>
<span><span class="co">## [121] grid_4.3.3              vctrs_0.6.5             slam_0.1-50            </span></span>
<span><span class="co">## [124] promises_1.3.0          xtable_1.8-4            cluster_2.1.6          </span></span>
<span><span class="co">## [127] evaluate_0.23           magick_2.8.3            cli_3.6.2              </span></span>
<span><span class="co">## [130] compiler_4.3.3          rlang_1.1.3             crayon_1.5.2           </span></span>
<span><span class="co">## [133] rngtools_1.5.2          simona_1.3.12           mclust_6.1.1           </span></span>
<span><span class="co">## [136] skmeans_0.2-16          plyr_1.8.9              fs_1.6.4               </span></span>
<span><span class="co">## [139] stringi_1.8.4           viridisLite_0.4.2       BiocParallel_1.34.2    </span></span>
<span><span class="co">## [142] munsell_0.5.1           Biostrings_2.68.1       lazyeval_0.2.2         </span></span>
<span><span class="co">## [145] GOSemSim_2.26.1         Matrix_1.6-5            patchwork_1.2.0        </span></span>
<span><span class="co">## [148] bit64_4.0.5             ggplot2_3.5.1           KEGGREST_1.40.1        </span></span>
<span><span class="co">## [151] shiny_1.8.1.1           highr_0.10              gridtext_0.1.5         </span></span>
<span><span class="co">## [154] igraph_2.0.3            memoise_2.0.1           bslib_0.7.0            </span></span>
<span><span class="co">## [157] ggtree_3.8.2            fastmatch_1.1-4         bit_4.0.5              </span></span>
<span><span class="co">## [160] downloader_0.4          ape_5.8                 gson_0.1.0</span></span></code></pre>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
